docs/ADR-72-ab-testing.md

```diff
@@ -0,0 +1,153 @@
+>
+> - assuming total: `1.000` users
+> - activation rollout (50%): `500` users
+> - `enabled` variant (50%): `250` users
+> - `disabled` variant (50%): `250` users
+>
+> For a total of `1.000` only half of then will count as testing users and half of those will see the `enabled` options, that mean that `750` users will see the old version but only `250` will be consider in the result of the A/B testing.
+
+Once your A/B testing is ready you should get something like this:
+
+```json5
+{
+  "flags": {
+    "test-flag": true, // not an A/B testing
+    "test-test-feature_name_variant": true,
+  },
+  "variants": {
+    "test-feature_name_variant": {
+      "name": "enabled" // or "disabled"
+    }
+  }
+}
+```
+
+If instead your A/B testing is inactive you should get something like this:
+
+```json5
+{
+  "flags": {
+    "test-flag": true, // not an A/B testing
+  },
+  "variants": {}
+}
+```
+
+#### Including extra data
+
+Some times you need to include additional data to know how to handle each variant, this is use full because you can develop the variant implementation and detached from the end value (that can be content that is not ready jet) you will be able to develop and deploy you a/b testing and other team in charge of produce the content will be able to test it without any other interaction with your code.
+
+![Extra data](./resources/ADR-70/006-extra-data.png)
+
+### Segment: measuring your new A/B testing
+
+To measure if any of the variants of an A/B testing is significant we need to know three thing:
+
+- the total number of user in the A/B testing
+- which variant got each one
+- the success metric
+
+To know the total number of users we send an event `feature_flags` each time a user gets a new set of feature flags, but to know which variant each user gets we need to format our feature flags as the following array:
+
+```js
+[
+  `${FEATURE_FLAG_NAME}`,
+  `${FEATURE_FLAG_NAME}:${FEATURE_FLAG_VARIANT_NAME}`,
+]
+```
+
+For example, the previous feature flags will be format as:
+
+```json5
+[
+  "test-flag", // not an A/B testing
+  "test-feature_name_variant",
+  "test-feature_name_variant:enabled",
+  // or "test-feature_name_variant:disabled",
+]
+```
+
+And send it in the prop `featureFlags` this way we can easy search feature flags and or variant.
+
+```ts
+analytics.track(
+  `feature_flags`,
+  {
+    featureFlags: [
+      "test-flag", // not an A/B testing
+      "test-feature_name_variant",
+      "test-feature_name_variant:enabled",
+    ]
+  }
+)
+```
+
+Finally you define your success metric and send attach the `featureFlags` prop to it.
+
+```ts
+analytics.track(
+  `user_enter_successfully`,
+  {
+    featureFlags: [
+      "test-flag", // not an A/B testing
+      "test-feature_name_variant",
+      "test-feature_name_variant:enabled",
+    ]
+  }
+)
+```
+
+Ideally you will send each event to segment with the `featureFlags` prop, that way you can change your success metric later.
```

Comment:
We don't need this either from the data team. We just need to get an event when the user first sees the experiment, we don't need the feature flag in any other event afterwards.